kube-client/src/client/builder.rs

```diff
@@ -82,6 +82,15 @@ impl TryFrom<Config> for ClientBuilder<GenericService> {
         let mut connector = HttpConnector::new();
         connector.enforce_http(false);
 
+        #[cfg(feature = "aws-lc-rs")]
+        {
+            if rustls::crypto::CryptoProvider::get_default().is_none() {
+                rustls::crypto::aws_lc_rs::default_provider()
+                    .install_default()
+                    .unwrap();
```

Comment:
I think our `Error` supports rustls errors: https://docs.rs/kube/latest/kube/enum.Error.html#variant.RustlsTls under a feature

Reply:
oops my bad, `?` is probably the way to go ^^

Reply:
setting the default provider returns the `Err(provider)` on error, I've added a new variant of `kube::Error`.